moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -237,6 +238,13 @@
 
   private int doPeek() throws IOException {
     int peekStack = stack[stackSize - 1];
+
+    if (peekStack == JsonScope.EMPTY_DOCUMENT
+            && source.request(BYTE_ORDER_MARK.size())
+            && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
```

Comment:
The reason to add it to moshi is so when people use moshi with retrofit (or
any other library on top of its reflection parsing), they don't have to
write manual parsers for everything.

On Mon, Jul 4, 2016, 22:42 Jake Wharton <notifications@github.com> wrote:

> In moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java
> <https://github.com/square/moshi/pull/170#discussion_r69507200>:
>
> > @@ -237,6 +238,13 @@
> >
> >    private int doPeek() throws IOException {
> >      int peekStack = stack[stackSize - 1];
> > +
> > +    if (peekStack == JsonScope.EMPTY_DOCUMENT
> > +            && source.request(BYTE_ORDER_MARK.size())
> > +            && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
>
> This was released in 1.9.0. I'm tempted to say that since it's so easy to
> test and skip the BOM that we don't really need to add it to Moshi.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/square/moshi/pull/170/files/c9fbc2edec57d29e20c57a981b66e805ac953b5e#r69507200>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACLkWtVwhTgUccnBmZaeGCFXHcuvvTIyks5qSeDPgaJpZM4IvaIS>
> .
>